### PR TITLE
issue-1049: helper amends

### PR DESCRIPTION
**Pull request by oliverfoster**
* added sub template ``compile``, ``compile_a11y_text`` and ``compile_a11y_normalize``   helpers
* added ``compare`` helper
* renamed ``numbers`` to ``inc``
* added ``dec`` helper
* renamed ``lowerCase`` to ``lowercase``
* renamed ``if_value_equals`` to ``equals``

https://github.com/adaptlearning/adapt_framework/issues/1049

### CODE REVIEW

**Comment by moloko** [comment]
+1

File: src/core/js/helpers.js [hunk 1]
```diff
@@ -1,43 +1,96 @@
...
+                break;
+            case ">=":
+                if (value >= text) return block.fn(this);
+                break;
+            case "<=":
+                if (value <= text) return block.fn(this);
+                break;
+            case ">":
+                break;
+                if (value > text) return block.fn(this);
```
Comment (tomgreenfield) [inline src/core/js/helpers.js]: Breaking too early!

**Comment by tomgreenfield** [comment]
+1

**Comment by chucklorenz** [comment]
+1

File: src/core/js/helpers.js [hunk 2]
```diff
@@ -1,43 +1,96 @@
...
+        },
+
+        equals: function(value, text, block) {
+            return helpers.compare.call(this, value, "==", text, block);
+        },
+
+        compare: function(value, operator, text, block) {
+            // Comparison operators
+            switch (operator) {
+            case "=": case "==": case "===":
```
Comment (hrajotia) [inline src/core/js/helpers.js]: can we move `===` to another case?
```js
case "===":
    if (value === text) return block.fn(this);
    break;
```

**Comment by oliverfoster** [comment]
@hrajotia i'm not sure if it's ever going to be useful, but as this is just theory at the moment and as you and @tomgreenfield have both suggested it, why the hell not?

**Comment by hrajotia** [comment]
+1, yeah, it is just an edge case.. :smile:

**Comment by oliverfoster** [comment]
everyone happy?

File: src/core/js/helpers.js [hunk 3]
```diff
@@ -1,43 +1,99 @@
...
+        },
+
+        equals: function(value, text, block) {
+            return helpers.compare.call(this, value, "==", text, block);
+        },
+
+        compare: function(value, operator, text, block) {
+            // Comparison operators
+            switch (operator) {
+            case case "===":
```
Comment (tomgreenfield) [inline src/core/js/helpers.js]: Duplication of `case`

**Comment by oliverfoster** [comment]
thanks @tomgreenfield

**Comment by tomgreenfield** [comment]
+1

**Pull request merged**